source/Set.h

```diff
@@ -65,6 +69,17 @@ const Type *Set<Type>::Find(const std::string &name) const
 
 
 
+template <class Type>
+const Type *Set<Type>::Sample() const
+{
+	if(this->size() == 0)
```

Comment:
I have a question regarding this.
Why '== 0' and not '<= 0' instead? Should '< 0' cases be allowed? (Is it even possible in theory?)
If not allowed, is there any reason against changing it to '<= 0' for extra safety? Or doesn't it matter?

Reply:
It’s just an early return guard for performance. Negative and positive numbers result in action.

Reply:
At least that is what I remember. Will look deeper later.